elfpy/eth/errors/errors.py

```diff
@@ -0,0 +1,60 @@
+"""Error handling for the hyperdrive ecosystem"""
+from typing import Literal, Sequence, TypedDict
+
+from eth_utils.conversions import to_hex
+from eth_utils.crypto import keccak
+from web3.contract.contract import Contract
+from web3.types import ABIFunctionParams
+
+
+def decode_error_selector_for_contract(error_selector: str, contract: Contract) -> str:
+    """Decode the error selector for a contract,
+
+    Arguments
+    ---------
+
+    error_selector : str
+        A 3 byte hex string obtained from a keccak256 has of the error signature, i.e.
+        'InvalidToken()' would yield '0xc1ab6dc1'.
+    contract: Contract
+        A web3.py Contract interface, the abi is required for this function to work.
+
+    Returns
+    -------
+    str
+       The name of the error. If the error is not found, returns UnknownError.
+    """
+
+    abi = contract.abi
+    if not abi:
```

Comment:
I think we want to explicitly say `if abi is None` to avoid unexpexted truthy issues